x-pack/plugins/fleet/server/services/agents/crud.ts

```diff
@@ -138,7 +139,7 @@ export async function getAgentsByKuery(
 
   return {
     agents,
-    total: agents.length,
+    total: res.body.hits.total.value,
```

Comment:
The way I interpret lines 134-138 are that if `showUpgradeable` is true, `agents` will be narrowed to only include upgradeable agents.

What would this return in that case? It appears the count will before the full/initial list, but `agents` would be a list of only upgradeable agents. i.e. potentially an array of 3 items but a `total: 5`.

It's totally possible I misunderstand what's happening here. Thanks for researching and sending a PR.

Reply:
`showUpgradeable` is a little weird here as the filtering is happening when we get the results,I am wondering if we can implement the show upgradeable as a filter in the ES query it will make things easier, otherwise I guess it's ok to have an inaccurate total for the show upgradeable filter an create an issue for that.